inspectit-gepard-agent/src/main/java/rocks/inspectit/gepard/agent/transformation/TransformationManager.java

```diff
@@ -34,7 +36,16 @@ public AgentBuilder modify(AgentBuilder agentBuilder) {
     DynamicTransformer transformer = new DynamicTransformer(instrumentationState);
     InspectitListener listener = new InspectitListener();
 
-    // In the future, we might add a white- or black-list for types
-    return agentBuilder.type(any()).transform(transformer).with(listener);
+    return agentBuilder.type(typeMatcher()).transform(transformer).with(listener);
+  }
+
+  /**
+   * Defines all types, which should (or should not) be transformed. We don't want to transform our
+   * own classes.
+   *
+   * @return the type matcher for transformation
+   */
+  private ElementMatcher.Junction<TypeDescription> typeMatcher() {
+    return not(nameStartsWith("rocks.inspectit"));
```

Comment:
I think you should use rocks.inspectit.gepard.agent here.
Otherwise, the agent would not be allowed to instrument the config-server (?)

Reply:
True